### PR TITLE
Last of linting

### PR DESCRIPTION
**Pull request by katerberg**
## What changes did you make?

- Pulls the unstable components out of the top level component
- This is definitely not pretty and there's a lot I'd probably do differently if we started from scratch, but it at least makes these more manageable even if a lot of this stuff could/should be pulled out to hooks and helpers.

## Is there a ticket that you are fixing?

[*Please link it here if so*](https://gitlab.com/jklabsinc/opslevel-bs/-/issues/23)

## Changelog

- [x] I have added a Changie entry or given a reason why this does not need an entry in this section.

Shar

## Screenshots

Shared in slack to avoid dumping all of my tabs

### CODE REVIEW

File: src/components/ServiceMaturityReport.tsx [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+  LevelCategory,
+  LevelCheckResults,
+  LevelNode,
+  OpsLevelService,
+  OverallLevel,
+  ScorecardStats,
+} from "../types/OpsLevelData";
+
+type Props = {
+  exportEntity: (event: React.MouseEvent) => void;
```
Comment (patrick-th) [inline src/components/ServiceMaturityReport.tsx:18]: should this be onExportRequested or something?

File: src/components/ServiceMaturityReport.tsx [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+  exporting: boolean;
+  levels: LevelNode[];
+  service: OpsLevelService;
+  checkResultsByLevel: LevelCheckResults[] | undefined;
+  selectedCategories: string[];
+  scorecards: ScorecardStats[] | undefined;
+  overallLevel?: OverallLevel;
+  setSelectedCategories: (selectedCategories: string[]) => void;
```
Comment (patrick-th) [inline src/components/ServiceMaturityReport.tsx:26]: onCategorySelectionChange?

**Comment by patrick-th** [comment]
I didn't go super deep but it makes sense on a high level - we'll have a final look at + test of the end product of all this when merging the linting branch anyway. Left minor comments about event property naming.

**Review by patrick-th — APPROVED** [review]

File: src/components/ServiceMaturityReport.tsx (hunk 1 above)
Comment (katerberg) [reply src/components/ServiceMaturityReport.tsx:18]: Yeah, reasonable. I was trying to balance "change as little as possible" with "do it right", but happy to do it right!

Updating now.

File: src/components/ServiceMaturityReport.tsx (hunk 2 above)
Comment (katerberg) [reply src/components/ServiceMaturityReport.tsx:26]: Yeah, reasonable. I was trying to balance "change as little as possible" with "do it right", but happy to do it right!

Updating now.

**Pull request merged**